back/spec/services/slug_service_spec.rb

```diff
@@ -2,47 +2,63 @@
-      unpersisted_users = [
-        build(:user, first_name: 'jose'),
-        build(:user, first_name: 'jose')
+    it 'generates unique slugs for unpersisted records with the same field values used for slugs' do
+      unpersisted_records = [
+        build(:project, title_multiloc: { 'en' => 'A Project' }),
+        build(:project, title_multiloc: { 'en' => 'A Project' })
       ]
-      expect(service.generate_slugs(unpersisted_users, &:first_name)).to eq %w[
-        jose
-        jose-1
+
+      expect(service.generate_slugs(unpersisted_records){|r| r.title_multiloc.values.first}).to eq %w[
+        a-project
+        a-project-1
       ]
     end
 
-    it 'generates unique slugs when one existing user already has the slug' do
-      _persisted_user = create(:user, slug: 'jose')
-      unpersisted_users = [
-        build(:user, first_name: 'jose'),
-        build(:user, first_name: 'jose')
+    it 'generates unique slugs when one existing record already has the slug' do
+      _persisted_record = create(:project, slug: 'a-project')
+      unpersisted_records = [
+        build(:project, title_multiloc: { 'en' => 'A Project' }),
+        build(:project, title_multiloc: { 'en' => 'A Project' })
       ]
-      expect(service.generate_slugs(unpersisted_users, &:first_name)).to eq %w[
-        jose-1
-        jose-2
+
+      expect(service.generate_slugs(unpersisted_records){|r| r.title_multiloc.values.first}).to eq %w[
+        a-project-1
+        a-project-2
       ]
     end
 
-    it 'generates unique slugs when existing users already have the slug' do
-      _persisted_users = [
-        create(:user, slug: 'jose'),
-        create(:user, slug: 'jose-1'),
-        create(:user, slug: 'paulo-18')
+    it 'generates unique slugs when existing records already have the slug' do
+      _persisted_records = [
+        create(:project, slug: 'a-project'),
+        create(:project, slug: 'a-project-1'),
+        create(:project, slug: 'another-project-18')
       ]
-      unpersisted_users = [
-        build(:user, first_name: 'jose'),
-        build(:user, first_name: 'paulo')
+      unpersisted_records = [
+        build(:project, title_multiloc: { 'en' => 'A Project' }),
+        build(:project, title_multiloc: { 'en' => 'Another Project' })
       ]
-      expect(service.generate_slugs(unpersisted_users, &:first_name)).to eq %w[
-        jose-2
-        paulo-19
+
+      expect(service.generate_slugs(unpersisted_records){|r| r.title_multiloc.values.first}).to eq %w[
+        a-project-2
+        another-project-19
       ]
     end
   end
 
+  describe 'generate_slug' do
+    it 'generates a unique slug when an existing record already has the slug' do
+      _persisted_record = create(:project, slug: 'a-project')
+      unpersisted_record = build(:project, title_multiloc: { 'en' => 'A Project' })
+
+      expect(service.generate_slug(unpersisted_record, unpersisted_record.title_multiloc.values.first)).to eq 'a-project-1'
+    end
+  end
```

Comment:
I added these tests to explain and demonstrate how the related method works, but it feels a little strange to illustrate with anything other than a collection of users, since we currently only ever use this method, via a call from `UserSlugService` when creating slugs for a collection of users.

More evidence that the related method should actually be moved int `UserSlugService`?